### PR TITLE
LDAP authentication support

### PR DESCRIPTION
**Pull request by alexjfisher**
For those who want to implement
https://docs.pulpproject.org/user-guide/authentication.html#ldap-whole-api-example

### CODE REVIEW

File: spec/classes/pulp_apache_spec.rb [hunk 1]
```diff
@@ -393,6 +393,26 @@
...
+      let :pre_condition do
+        "class {'pulp':
+           ldap_url           => 'ldaps://ad.example.com?sAMAccountName',
+           ldap_bind_dn       => 'cn=pulp,dc=example,dc=com',
+           ldap_bind_password => 'BIND_PASSWORD',
+          }"
+      end
+
+      it 'should configure apache for LDAP authentication' do
+        is_expected.to contain_concat__fragment('pulp-https-directories')
```
Comment (ekohl) [inline spec/classes/pulp_apache_spec.rb]: We have a ```verify_concat_fragment_exact_contents``` helper for this.

Comment (alexjfisher) [inline spec/classes/pulp_apache_spec.rb]: I've used the other helper `verify_concat_fragment_contents`.  Hope that's ok.

File: templates/ldap_custom_fragment.erb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    SetEnvIfNoCase ^Authorization$ "Basic.*" USE_APACHE_AUTH=1
+    Order allow,deny
+    Allow from env=!USE_APACHE_AUTH
+    Satisfy Any
+
+    # configure basic auth
+    AuthType basic
+    AuthBasicProvider ldap
+    AuthName "Pulp"
+    AuthLDAPURL "<%= scope['pulp::ldap_url'] %>"
```
Comment (ekohl) [inline templates/ldap_custom_fragment.erb]: Isn't this possible to pass in as ```auth_ldap_url``` to the directories options?

Comment (alexjfisher) [inline templates/ldap_custom_fragment.erb]: It was possible to configure some lines with directories options, but not others.  custom_fragments are added at the end of the template.  Mixing custom_fragment with other options ended up quite messy.  At least this way, the output matches the [documented ](https://docs.pulpproject.org/user-guide/authentication.html#ldap-whole-api-example)way of configuring this feature.

Comment (ekohl) [inline templates/ldap_custom_fragment.erb]: That makes sense and I suspected it was something like this.

**Comment by stbenjam** [comment]
It looks OK to me, I don't have an easy way to test the actual functionality at the moment though.

**Comment by alexjfisher** [comment]
Not sure what to suggest.  Are you happy that the PR doesn't change module behaviour unless an `ldap_url` is provided?  I've been using this PR with active directory for the last few weeks.

**Comment by ekohl** [comment]
I'm happy to accept it. The tests show it's the same as the documentation, doesn't change existing behaviour and I trust @alexjfisher's use in production.

**Comment by ekohl** [comment]
Thanks!

**Pull request merged**